src/semidiscretization.jl

```diff
@@ -159,59 +162,82 @@ end
-  boundary_conditions::BoundaryConditions
+  boundary_condition::BoundaryCondition
   source_terms::SourceTerms
   solver::Solver
   cache::Cache
   performance_counter::PerformanceCounter
 
-  function SemidiscretizationHyperbolic{Mesh, Equations, InitialConditions, BoundaryConditions, SourceTerms, Solver, Cache}(
+  function SemidiscretizationHyperbolic{Mesh, Equations, InitialConditions, BoundaryCondition, SourceTerms, Solver, Cache}(
       mesh::Mesh, equations::Equations,
-      initial_conditions::InitialConditions, boundary_conditions::BoundaryConditions,
+      initial_condition::InitialConditions, boundary_condition::BoundaryCondition,
       source_terms::SourceTerms,
-      solver::Solver, cache::Cache) where {Mesh, Equations, InitialConditions, BoundaryConditions, SourceTerms, Solver, Cache}
+      solver::Solver, cache::Cache) where {Mesh, Equations, InitialConditions, BoundaryCondition, SourceTerms, Solver, Cache}
     @assert ndims(mesh) == ndims(equations)
 
     performance_counter = PerformanceCounter()
 
-    new(mesh, equations, initial_conditions, boundary_conditions, source_terms, solver, cache, performance_counter)
+    new(mesh, equations, initial_condition, boundary_condition, source_terms, solver, cache, performance_counter)
   end
 end
 
 """
-    SemidiscretizationHyperbolic(mesh, equations, initial_conditions, solver;
+    SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver;
                                  source_terms=nothing,
-                                 boundary_conditions=nothing)
+                                 boundary_condition=nothing)
 
 Construct a semidiscretization of a hyperbolic PDE.
 """
-function SemidiscretizationHyperbolic(mesh, equations, initial_conditions, solver;
+function SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver;
                                       source_terms=nothing,
-                                      boundary_conditions=nothing, RealT=real(solver))
+                                      boundary_condition=nothing, RealT=real(solver))
 
   cache = create_cache(mesh, equations, solver, RealT)
+  _boundary_condition = digest_boundary_condition(boundary_condition)
+
+  SemidiscretizationHyperbolic{typeof(mesh), typeof(equations), typeof(initial_condition), typeof(_boundary_condition), typeof(source_terms), typeof(solver), typeof(cache)}(
+    mesh, equations, initial_condition, _boundary_condition, source_terms, solver, cache)
+end
+
+
+# allow passing named tuples of BCs constructed in an arbitrary order
+digest_boundary_condition(boundary_condition) = boundary_condition
 
-  SemidiscretizationHyperbolic{typeof(mesh), typeof(equations), typeof(initial_conditions), typeof(boundary_conditions), typeof(source_terms), typeof(solver), typeof(cache)}(
-    mesh, equations, initial_conditions, boundary_conditions, source_terms, solver, cache)
+function digest_boundary_condition(boundary_condition::NamedTuple{<:NTuple{2,Symbol}}) # 1D
```

Comment:
Improved in 6538246
```julia
julia> typeof( (a=1, b=2) )
NamedTuple{(:a, :b),Tuple{Int64,Int64}}
```
Now, we use the type of the second type parameter of `NamedTuple`, which is a type instead of a value and makes it easier to think about what's going on.

Reply:
Yes, thanks. But still - why is `<:` for a concrete type legal and/or necessary in the first place? Couldn't you just have written `NamedTuple{NTuple{2,Symbol}}`? Please note I am not trying to annoy, just to understand Julia better...

Reply:
```
julia> Float64 <: Float64
true
```
A type is always a subtype of itself. The first part for named tuples is particularly tricky since it's using a value as type parameter instead of a type. In fact, it didn't always work as expected, so it's great that you asked - now, it's fixed and should work as expected.

Reply:
Thanks for the lessen!